### PR TITLE
run-checks, tests/lib/snaps/: shellcheck fixes

### PR DESCRIPTION
**Pull request by bboozzoo**
Small series introducing fixes for warnings brought up by shellcheck 0.4.6.

### CODE REVIEW

File: run-checks [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (mvo5) [inline run-checks]: Curious, what bash specific feature are we using in here?

Comment (bboozzoo) [reply run-checks]: `            echo "- $snap_yaml"`

Apparently this is ambiguous, as `-` is used to denote flags. Shellcheck raises this as a warning with /bin/sh. It's still ambiguous wrt. the flags, but at least shellcheck is now silent.

IMO a proper, unambiguous version would be `printf "- $snap_yaml\n"`, but that looks just awkward.

Comment (mvo5) [reply run-checks]: I slightly prefer printf over making this bash specific. Yes, its slightly more ugly and a bit unusual but given how non-portable echo is, maybe the world needs to adjust to seeing more printf in shell scripts :)

Comment (bboozzoo) [reply run-checks]: Ok. FWIW here's how `echo` behaves (note that `%` is just zsh way of saying there's no newline):

```
maciek@corsair:~ /bin/echo "-n" "foo"
foo%
maciek@corsair:~ /bin/echo "-N" "foo"
-N foo
```

Comment (bboozzoo) [reply run-checks]: Pushed a workaround. We can avoid using printfs and still keep the offending pieces readable.

Comment (mvo5) [reply run-checks]: Nice! Now we can revert the first line back to /bin/sh, right?

**Pull request merged**